lib/babysitter.ex

```diff
@@ -0,0 +1,27 @@
+defmodule BabySitter do
+
+  alias BabySitter.ScheduleValidation, as: ScheduleValidation
+  alias BabySitter.BeforeBedTime, as: BeforeBedTime
+  alias BabySitter.Bedtime, as: BedTime
+  alias BabySitter.AfterMidnight, as: AfterMidnight
+
+  @too_early_message "Babysitter cannot start before 5:00PM"
+  @too_late_message "Babysitter must leave by 4:00AM"
+
+  def calculate_pay(start_time, hours, bed_time) do
+    cond do
+      ScheduleValidation.end_time_too_late?(start_time, hours) -> {:error, @too_late_message}
+      ScheduleValidation.start_time_too_early?(start_time) -> {:error, @too_early_message}
+      true -> {:ok, calculate_by_pay_rate(start_time, hours, bed_time)}
+    end
```

Comment:
Nice use of cond. I don’t see too many good uses for it, but it solves this nicely.